packages/tic-tac-toe/contracts/tests/tic-tac-toe.test.ts

```diff
@@ -0,0 +1,293 @@
+
+dotEnvExtended.load();
+
+jest.setTimeout(20000);
+
+const testProvider = new ethers.providers.JsonRpcProvider(
+  `http://localhost:${process.env.GANACHE_PORT}`
+);
+
+const PositionIndex = {
+  Start: PositionType.Start,
+  XPlaying: PositionType.XPlaying,
+  OPlaying: PositionType.OPlaying,
+  Draw: PositionType.Draw,
+  Victory: PositionType.Victory
+};
+
+let TicTacToe: Contract;
+
+const numParticipants = 2;
+const addresses = {
+  // participants
+  A: randomExternalDestination(),
+  B: randomExternalDestination()
+};
+
+export function setupContracts(
+  provider: ethers.providers.JsonRpcProvider,
+  artifact,
+  address
+): ethers.ethers.Contract {
+  const signer = provider.getSigner(0);
+
+  const contract = new ethers.Contract(address, artifact.abi, signer);
+  return contract;
+}
+
+async function sendTransaction(
+  contractAddress: string,
+  transaction: TransactionRequest
+): Promise<void> {
+  // move to devtools
+  const signer = testProvider.getSigner();
+  const response = await signer.sendTransaction({to: contractAddress, ...transaction});
+  await response.wait();
+}
+
+beforeAll(async () => {
+  TicTacToe = await setupContracts(
+    testProvider,
+    TicTacToeArtifact,
+    process.env.TTT_CONTRACT_ADDRESS
+  );
+});
+
+//      0  |  1  |  2
+//   +-----------------+
+//      3  |  4  |  5
+//   +-----------------+
+//      6  |  7  |  8
+
+// Draw
+//      X0  |  O5  |  X4
+//   +-----------------+
+//      X2  |  O1  |  O7
+//   +-----------------+
+//      O3  |  X6  |  X8
+
+const drawX = 0b101100011;
+const drawO = 0b010011100;
+
+// Binary representations of numbers
+const b = {
+  0: 0b000000000,
+  1: 0b000000001,
+  2: 0b000000010,
+  3: 0b000000011,
+  5: 0b000000101,
+  6: 0b000000110,
+  7: 0b000000111,
+  8: 0b000001011,
+  11: 0b000001011,
+  24: 0b000011000,
+  156: drawO,
+  354: 0b101100010,
+  355: drawX
+};
+
+// const x = 0b101100010;
+
+const balanceStartToAWins = [
+  {A: 5, B: 5},
+  {A: 6, B: 4}
+];
+
+const balanceBWinsToAWins = [
+  {A: 4, B: 6},
+  {A: 6, B: 4}
+];
+
+const balanceAWinsToBWins = [
+  {A: 6, B: 4},
+  {A: 4, B: 6}
+];
+
+const balanceBWinsToDraw = [
+  {A: 4, B: 6},
+  {A: 5, B: 5}
+];
+
+const balanceDrawToStart = [
+  {A: 5, B: 5},
+  {A: 5, B: 5}
+];
+
+const balanceAWinsToStart = [
+  {A: 6, B: 4},
+  {A: 6, B: 4}
+];
+
+const balanceBWinsToStart = [
+  {A: 4, B: 6},
+  {A: 4, B: 6}
+];
+
+const startToXPlaying = ['Start', 'XPlaying'];
+const xPlayingToOPlaying = ['XPlaying', 'OPlaying'];
+const oPlayingToXPlaying = ['OPlaying', 'XPlaying'];
+const xPlayingToVictory = ['XPlaying', 'Victory'];
+const oPlayingToVictory = ['OPlaying', 'Victory'];
+const oPlayingToDraw = ['OPlaying', 'Draw'];
+const drawToStart = ['Draw', 'Start'];
+const victoryToStart = ['Victory', 'Start'];
+
+describe('validTransition', () => {
+  it.each`
+    isValid  | positionType          | stake     | Xs                  | Os                  | balances               | description
```

Comment:
Nit: I quite like to have all my `true` cases together and all of my `false` cases together 🤓